### PR TITLE
Add Swagger document for all APIs

### PR DESCRIPTION
**Pull request by PeterYHChen**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -12,6 +13,22 @@
...
+# Configure Swagger before initilaizing it
+app.config['SWAGGER'] = {
+    "swagger_version": "2.0",
+    "specs": [
+        {
+            "version": "1.0.0",
+            "title": "Wishlist App",
+            "description": "The wishlists resource allow customers to create a collection of products that they wish they had the money to purchase.",
+            "endpoint": 'v1_spec',
+            "route": '/v1/spec'
```
Comment (brentdur) [inline server.py]: do we need/want `/v1` before `/spec` just wondering as we don't use versioning anywhere else

File: server.py [hunk 2]
```diff
@@ -29,12 +46,43 @@ def bad_request(error):
...
     return jsonify(status=400, error='Bad Request', message=message), 400
 
+######################################################################
+#  Index
+######################################################################
+@app.route("/")
+def index():
+    return jsonify(name='Wishlist REST API Service',
+                   version='1.0',
+                   docs=request.base_url + 'apidocs/index.html'), status.HTTP_200_OK
```
Comment (brentdur) [inline server.py]: does this load when I open `/` or do I open `apidocs`? Just confused about the syntax

File: server.py (hunk 1 above)
Comment (aashishdugar) [reply server.py]: I think the professor siad it was a good idea to include versioning in the urls.

Comment (aashishdugar) [reply server.py]: said*

File: server.py (hunk 2 above)
Comment (PeterYHChen) [reply server.py]: You open /apidoc/index.html. The / indicates which link for api docs

File: server.py (hunk 1 above)
Comment (PeterYHChen) [reply server.py]: I copied from sample code. Keep it just in case.

**Pull request merged**